src/gui/properties/peerlistwidget.cpp

```diff
@@ -400,9 +415,15 @@ void PeerListWidget::updatePeer(const QString &ip, const BitTorrent::TorrentHand
 
 void PeerListWidget::handleResolved(const QString &ip, const QString &hostname)
 {
-    QStandardItem *item = m_peerItems.value(ip, nullptr);
-    if (item)
-        item->setData(hostname, Qt::DisplayRole);
+    for (auto i = m_peerItems.cbegin(); i != m_peerItems.cend(); ++i) {
+        if (i.key().address.ip.toString() == ip) {
+            if (i.value()) {
```

Comment:
>Also it seems the nullptr check is not needed (please test that).

Qt docs say that `QHash::value()` returns a https://doc.qt.io/qt-5/containers.html#default-constructed-value i.e '0' for pointers so.. i guess it's needed.

Reply:
>Qt docs say that QHash::value() returns a https://doc.qt.io/qt-5/containers.html#default-constructed-value i.e '0' for pointers 

You misunderstand the `i` type, it is [`QHash::iterator`](https://doc.qt.io/qt-5/qhash-iterator.html#value) not `QHash`.

>i guess it's needed.

I still think otherwise because we always store valid `QStandardItem *` in `m_peerItems` and should not be not possible to get a `nullptr`.

Also since `i` is actually an iterator I would rename it to `iter` to avoid misunderstanding, but it is up to you.

Reply:
> the i type, it is QHash::iterator

Oops... okay i'll remove the check and change to iter.